soc/nordic/nrf54l/soc.c

```diff
@@ -149,7 +151,15 @@ static int nordicsemi_nrf54l_init(void)
 		nrf_power_task_trigger(NRF_POWER, NRF_POWER_TASK_CONSTLAT);
 	}
 
+#if defined(CONFIG_SOC_NRF54L_VREG_MAIN_DCDC)
 	if (IS_ENABLED(CONFIG_SOC_NRF54L_VREG_MAIN_DCDC)) {
+#elif (DT_PROP(DT_INST(0, nordic_nrf5x_regulator), regulator_initial_mode) == \
+		NRF5X_REG_MODE_DCDC)
+	{
+#else
+	/* DCDC is not configured neither in KConfig nor DTS. Disable its configuration. */
+	if (0) {
+#endif
```

Comment:
please, split these changes into its own commit, also, `if (0)` is redundant, just ifdef out the whole block